Lib/test/test_fstring.py

```diff
@@ -980,11 +986,11 @@ def test_roundtrip_raw_quotes(self):
         self.assertEqual(fr'\"\'\"\'', '\\"\\\'\\"\\\'')
 
     def test_fstring_backslash_before_double_bracket(self):
-        self.assertEqual(f'\{{\}}', '\\{\\}')
-        self.assertEqual(f'\{{', '\\{')
-        self.assertEqual(f'\{{{1+1}', '\\{2')
-        self.assertEqual(f'\}}{1+1}', '\\}2')
-        self.assertEqual(f'{1+1}\}}', '2\\}')
+        self.assertEqualWithDeprecationWarning(r"f'\{{\}}'", '\\{\\}')
+        self.assertEqualWithDeprecationWarning(r"f'\{{'", '\\{')
+        self.assertEqualWithDeprecationWarning(r"f'\{{{1+1}'", '\\{2')
+        self.assertEqualWithDeprecationWarning(r"f'\}}{1+1}'", '\\}2')
+        self.assertEqualWithDeprecationWarning(r"f'{1+1}\}}'", '2\\}')
```

Comment:
If you used a `subTest` like this, you wouldn't have to define the `assertEqualWithDeprecationWarning` method:

```suggestion
        deprecated_cases = [
            (r"f'\{{\}}'",   '\\{\\}'),
            (r"f'\{{'",      '\\{'),
            (r"f'\{{{1+1}'", '\\{2'),
            (r"f'\}}{1+1}'", '\\}2'),
            (r"f'{1+1}\}}'", '2\\}')
        ]
        for case, expected_result in deprecated_cases:
            with self.subTest(case=case, expected_result=expected_result):
                with self.assertWarns(DeprecationWarning):
                    result = eval(case)
                self.assertEqual(result, expected_result)
```